src/http/client/response.cr

```diff
@@ -31,6 +31,10 @@ class HTTP::Client::Response
     @body
   end
 
+  def successful?
```

Comment:
I think anything that is not `success?` is a failure, so having `failed?` or `fail?`, as you can do:

```crystal
unless response.success?
  # uh-oh, something has failed! ;'(
end
```

Then it gets tricky for `error?` (is a server side error? or a client side error like bad request?)

Same goes for `redirect?` is a permanent one, a temporary one? I think will be overwhelming to have many helper methods for those cases, so perhaps a generic one is OK.

Cheers.